lib/mix/ecto.ex

```diff
@@ -86,12 +86,42 @@ defmodule Mix.Ecto do
 
   @doc """
   Asks if the user wants to open a file based on ECTO_EDITOR.
+
+  By default, it attempts to open the file and line using the
+  `file:line` notation. For example, if your editor is called
+  `subl`, it will open the file as:
+
+      subl path/to/file:line
+
+  It is important that you choose an editor command that does
+  not block nor that attempts to run an editor directly in the
+  terminal. Command-line based editors likely need extra
+  configuration so they open up the given file and line in a
+  separate window.
+
+  Custom editors are supported by using the `__FILE__` and
+  `__LINE__` notations, for example:
+
+      ECTO_EDITOR="my_editor +__LINE__ __FILE__"
+
+  and Elixir will properly interpolate values.
+
   """
-  @spec open?(binary) :: boolean
-  def open?(file) do
+  @spec open?(binary, non_neg_integer) :: boolean
+  def open?(file, line \\ 1) do
     editor = System.get_env("ECTO_EDITOR") || ""
+
     if editor != "" do
-      :os.cmd(to_charlist(editor <> " " <> inspect(file)))
+      command =
+        if editor =~ "__FILE__" or editor =~ "__LINE__" do
+          editor
+          |> String.replace("__FILE__", inspect(file))
+          |> String.replace("__LINE__", Integer.to_string(line))
+        else
+          "#{editor} #{inspect(file)}:#{line}"
+        end
+
+      :os.cmd(String.to_charlist(command))
```

Comment:
I couldn't find a simple way to test it.

Reply:
I'd do it the same way IEx.Helpers.open/1 is tested: https://github.com/elixir-lang/elixir/blob/b662d32fc3171537778aa896a0b4fd65af06d354/lib/iex/test/iex/helpers_test.exs#L147-L156

Reply:
I tried using something similar but it doesn't seem to work. I don't know if I'm missing something. 

```
ExUnit.CaptureIO.capture_io(fn ->
  :os.cmd('echo "something"')
end)
``` 
![image](https://user-images.githubusercontent.com/13632762/105252044-7a7b8900-5b5b-11eb-94dc-81688c43573d.png)

I tried a different approach by creating a file in tmp directory. WDYT?

Reply:
Hey, I was just curious and looked this up. So `:os.cmd()` already captures the output and returns it as a string. That's why `capture_io()` doesn't return anything.

You could actually write the string to stdout, like so: `:os.cmd(to_charlist(command)) |> IO.write()`
That should make the test pass :+1:. The echo command will return a trailing newline btw, e.g.: `"something\n"`